src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SymbolStore;
+using Microsoft.SymbolStore.KeyGenerators;
+
+namespace Microsoft.SymbolManifestGenerator
+{
+    public class SymbolManifestGenerator
+    {
+        private const int Private = 340;
+        public SymbolManifestGenerator()
+        {
+        }
+
+        public static void GenerateManifest(ITracer tracer, DirectoryInfo dir, string manifestFileName)
+        {
+            try
+            {
+                ManifestDataV1 manifestData = new();
+
+                // Iterate through all files recursively in the directory
+                foreach (FileInfo file in dir.GetFiles("*", SearchOption.AllDirectories))
+                {
+                    string relativePath = file.DirectoryName.Replace(dir.FullName, string.Empty).TrimStart(Path.DirectorySeparatorChar);
+                    FileStream fileStream = file.OpenRead();
+                    SymbolStoreFile symbolStoreFile = new(fileStream, file.FullName);
+                    FileKeyGenerator generator = new(tracer, symbolStoreFile);
+                    IEnumerable<SymbolStoreKey> clrKeys = generator.GetKeys(KeyTypeFlags.ClrKeys);
+                    foreach (SymbolStoreKey clrKey in clrKeys)
+                    {
+                        string additionalDebugEntryFileName = clrKey.FullPathName;
+                        string basedirRelativePath = Path.Combine(relativePath, additionalDebugEntryFileName);
+                        string additionalDebugEntryFilePath = Path.Combine(dir.FullName, basedirRelativePath);
+                        FileInfo additionalDebugEntryFileInfo = new(additionalDebugEntryFilePath);
+                        if (!File.Exists(additionalDebugEntryFilePath))
+                        {
+                            tracer.Warning($"File {basedirRelativePath} does not exist. Skipping.");
```

Comment:
Should this be an error or should it skip the file?  Not sure.  It doesn't actually skipping anything in this current code.

Reply:
Ah forgot to add continue. I think this should be a warning and not cancel the manifest generation.

The ClrKeys loop from `generator.GetKeys(KeyTypeFlags.ClrKeys);` is the "known" set of special files for a runtime module (clr.dll/coreclr.dll), so it only checks if those file paths (path of directory passed to the API + relative path of runtime module sans filename + special file name) exist. If the known file doesn't exist at that path, not only would we not be able to generate a filehash for validation, but excluding it from the manifest wouldn't affect symbol uploading, as there wouldn't have been the special file to upload in the first place.

I think skipping would be more lenient than cancelling the generation, unless we want to guarantee that all the "known" possible special files will *always* be there